pkg/controller/servicemesh/webhookca/controller.go

```diff
@@ -49,47 +47,47 @@ var autoRegistrationMap = map[string]CABundleSource{
 		SecretNameKeyPairs: []SecretNameKeyPair{
 			{
 				SecretName: galleySecretName,
-				Key:        common.IstioRootCertKey,
+				Keys:       []string{common.IstioRootCertKey},
 			},
 		},
 	},
 	sidecarInjectorWebhookNamePrefix: &SecretCABundleSource{
 		SecretNameKeyPairs: []SecretNameKeyPair{
 			{
 				SecretName: sidecarInjectorSecretName,
-				Key:        common.IstioRootCertKey,
+				Keys:       []string{common.IstioRootCertKey},
 			},
 		},
 	},
 	istiodWebhookNamePrefix: &SecretCABundleSource{
 		SecretNameKeyPairs: []SecretNameKeyPair{
 			{
 				SecretName: istiodCustomCertSecretName,
-				Key:        common.IstiodCertKey,
+				Keys:       []string{common.IstiodCertKey, common.IstiodTLSSecretCertKey},
 			},
```

Comment:
Wouldn't creating two `SecretNameKeyPair` entries here instead of changing  `Key string` to `Keys []string` be a simpler approach? Or is the problem that you can't use the same `SecretName` in two entries (even if the `Key` is different)?

Reply:
> is the problem that you can't use the same SecretName in two entries

No, there is no obstacle to do it. I implemented this PR at night, so my brain wasn't bright enough... I will revert this change and will just add another secret name and key pair. It will significantly simplify this PR. Then I will be able to revert those nested loops with `break label`.

Reply:
Ah, yes. Welcome to fatherhood :)

That's exactly why I need people to review my own PRs. I make silly mistakes like this all the time.